app/history.py

```diff
@@ -19,44 +19,65 @@
 import app.log
 import cPickle as pickle
 import os
+import hashlib
+import time
+import app.prefs
 
-data = {}
-path = None
-
-def get(keyPath, default=None):
-  global data
-  cursor = data
-  for i in keyPath[:-1]:
-    cursor = cursor.setdefault(i, {})
-  return cursor.get(keyPath[-1], default)
-
-def set(keyPath, value):
-  global data
-  cursor = data
-  for i in keyPath[:-1]:
-    cursor = cursor.setdefault(i, {})
-  cursor[keyPath[-1]] = value
-  #assert get(keyPath) == value
-
-def loadUserHistory(filePath):
-  global data, path
-  try:
-    path = filePath
-    if os.path.isfile(path):
-      with open(path, "rb") as file:
-        data = pickle.load(file)
-    else:
-      data = {}
-  except Exception, e:
-    app.log.error('exception')
-    data = {}
+fileHistory = {}
```

Comment:
It looks like fileHistory expects that a single file will edited at a time. This is currently true, but I'd like to expand ci_edit to allow multiple files being edited. How about if the fileHistory was not stored as a global. Maybe it can be an object that gets created to load or save the values, but is not stored otherwise?